app/org/sagebionetworks/bridge/services/MpowerVisualizationService.java

```diff
@@ -0,0 +1,79 @@
+package org.sagebionetworks.bridge.services;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.commons.lang3.StringUtils;
+import org.joda.time.LocalDate;
+import org.joda.time.Period;
+import org.joda.time.PeriodType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import org.sagebionetworks.bridge.dao.MpowerVisualizationDao;
+import org.sagebionetworks.bridge.exceptions.BadRequestException;
+import org.sagebionetworks.bridge.json.DateUtils;
+
+/** mPower visualization service. */
+@Component
+public class MpowerVisualizationService {
+    private static final int MAX_RANGE_DAYS = 45;
+
+    private MpowerVisualizationDao mpowerVisualizationDao;
+
+    /** mPower visualization DAO. */
+    @Autowired
+    final void setMpowerVisualizationDao(MpowerVisualizationDao mpowerVisualizationDao) {
+        this.mpowerVisualizationDao = mpowerVisualizationDao;
+    }
+
+    /**
+     * <p>
+     * Get the visualization for the given date range. We use raw JSON to allow for rapid development. When we need to
+     * genericize and harden this, we can refactor this class as necessary.
+     * </p>
+     * <p>
+     * See https://sagebionetworks.jira.com/wiki/display/BRIDGE/mPower+Visualization for details.
+     * </p>
+     *
+     * @param healthCode
+     *         user's health code
+     * @param startDateStr
+     *         start date for visualization
+     * @param endDateStr
+     *         end date for visualization
+     * @return raw JSON containing visualization data
+     */
+    public JsonNode getVisualization(String healthCode, String startDateStr, String endDateStr) {
```

Comment:
I would expect types to be converted by the time a service is called. IMO the dates are only strings because they were retrieved via HTTP, they should be parsed in the controller and all HTTP conversion dealt with before the service. This allows the service to be used elsewhere without spurious conversions to strings.